pulser-core/pulser/register/base_register.py

```diff
@@ -77,6 +78,17 @@ def __init__(
             [pm.AbstractArray(v, dtype=float) for v in qubits.values()]
         )
         self._ids: tuple[QubitId, ...] = tuple(qubits.keys())
+        not_str = [id for id in self._ids if not isinstance(id, str)]
+        if not_str:
+            warnings.simplefilter("always")
+            warnings.warn(
+                "Usage of `int`s as `QubitId`s will be deprecated. Define "
```

Comment:
We are not actually checking that they are `int`, just that they are not str... To be precise, we should say
```suggestion
                "Usage of `int`s or other non-`str` types as `QubitId`s will be deprecated. Define "
```